lib/rest-api.php

```diff
@@ -206,6 +206,11 @@ function gutenberg_add_permalink_template_to_posts( $response, $post, $request )
 		return $response;
 	}
 
+	$post_type_obj = get_post_type_object( $post->post_type );
+	if ( ! is_post_type_viewable( $post_type_obj ) || ! $post_type_obj->public ) {
+		return $response;
+	}
```

Comment:
I pulled this logic [over from the 5.0 branch](https://github.com/WordPress/wordpress-develop/blob/edbb865da0410182a36695160abecb35885a4f4c/src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php#L1602).